packages/runtime/container-runtime/src/opTelemetry.ts

```diff
@@ -0,0 +1,61 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IDeltaManager } from "@fluidframework/container-definitions";
+import {
+    IDocumentMessage,
+    ISequencedDocumentMessage,
+} from "@fluidframework/protocol-definitions";
+import { isSystemMessage } from "@fluidframework/protocol-base";
+
+export class OpTracker {
+    /**
+     * Used for storing the message content size when
+     * the message is pushed onto the inbound queue.
+     */
+    private readonly messageSize = new Map<number, number>();
+    private _nonSystemOpCount: number = 0;
+    public get nonSystemOpCount(): number {
+        return this._nonSystemOpCount;
+    }
+
+    private _opsSizeAccumulator: number = 0;
+    public get opsSizeAccumulator(): number {
+        return this._opsSizeAccumulator;
+    }
+
+    public constructor(
+        deltaManager: IDeltaManager<ISequencedDocumentMessage, IDocumentMessage>,
+        disabled: boolean,
+    ) {
+        if (disabled) {
+            return;
+        }
+
+        // Record the message content size when we receive it.
+        // We should not log this value, as summarization can happen between the time the message
+        // is received and until it is processed (the 'op' event).
+        deltaManager.inbound.on("push", (message: ISequencedDocumentMessage) => {
+            // Some messages my already have string contents at this point,
+            // so stringifying them again will add inaccurate overhead.
+            const stringContents = typeof message.contents === "string" ?
+                message.contents :
+                JSON.stringify(message.contents);
```

Comment:
I'd expect all messages to be strings here. We can glance at the history (when this code was added in DeltaManager), but if it was added more than a year ago, I'd just remove all this back-compat code and assert that it's always a string.

Reply:
Some system messages are objects:

![image](https://user-images.githubusercontent.com/84357545/162230908-414a2f0d-6485-491a-898b-36f7b4f2a938.png)

I believe we need to measure their size also, right?

Reply:
ClientJoin & ClientLeave messages use `data`. Control messages also use it, but clients will never see those since they are not sequenced. 
Relevant code is [here](https://github.com/microsoft/FluidFramework/blob/11d8226e478519bd7d46563d8629f6a2008bec58/server/routerlicious/packages/lambdas/src/deli/lambda.ts#L809) and [here](https://github.com/microsoft/FluidFramework/blob/11d8226e478519bd7d46563d8629f6a2008bec58/server/routerlicious/packages/lambdas/src/deli/lambda.ts#L861).

The summary ack and nack case is odd. Scribe will set both the contents and data property for it ([code](https://github.com/microsoft/FluidFramework/blob/11d8226e478519bd7d46563d8629f6a2008bec58/server/routerlicious/packages/lambdas/src/scribe/lambda.ts#L499)). However the summaryAck & nack type are missing from deli's `extractDataContent` method so the data property is essentially ignored. Maybe someone meant to migrate those 2 types to use `data` instead of `contents` but didn't finish? That seems to be the case based on [this PR](https://github.com/microsoft/FluidFramework/pull/7231)

Reply:
Opened https://github.com/microsoft/FluidFramework/issues/9775